### ISSUE TITLE
Are MDX component overrides still supported?

### ISSUE THREAD

**Issue by alexpriceco** [storybookjs/storybook#9968]
### Bug description
I'm unable to use MDX component overrides, described in [theming.md](https://github.com/storybookjs/storybook/blob/next/addons/docs/docs/theming.md#mdx-component-overrides).

### To reproduce
Create some `document.stories.mdx` MDX document with a level 1 heading, then set `preview.js` to the following:
```jsx
import react from "react";
import { addParameters } from '@storybook/react';

const newHeader = ({ children }) => (
  <h1 style={{ color: "red" }}>TEST {children}</h1>
);

addParameters({
  docs: {
    components: {
      h1: newHeader,
    }
  }
})
```

### Expected behavior
All level 1 headers in MDX docs should have the word TEST before the content, and be red. This is not the case. It looks like the custom component isn't even being loaded.

<details><summary>System info</summary>
```
  System:
    OS: macOS 10.15.3
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
  Binaries:
    Node: 13.7.0 - /usr/local/bin/node
    Yarn: 1.21.1 - /usr/local/bin/yarn
    npm: 6.13.6 - /usr/local/bin/npm
  Browsers:
    Chrome: 80.0.3987.122
    Safari: 13.0.5
  npmPackages:
    @storybook/addon-actions: ^5.3.13 => 5.3.13 
    @storybook/addon-centered: ^5.3.12 => 5.3.13 
    @storybook/addon-docs: ^5.3.10 => 5.3.13 
    @storybook/addon-knobs: ^5.3.10 => 5.3.13 
    @storybook/react: 5.3.10 => 5.3.10 
    @storybook/theming: ^5.3.10 => 5.3.13 
```
</details>

**Comment by alexpriceco** [comment]
Bumping to remove inactive label.

**Comment by shilman** [comment]
sorry was looking into this and got distracted. i'll follow up.

**Comment by alexpriceco** [comment]
Thank you so much! I really appreciate you and the team’s efforts on Storybook. Absolutely no rush, and please stay safe in this difficult time.

**Comment by shilman** [comment]
I just tried it again in `official-storybook` and it seems to be working fine:

```js
// eslint-disable-next-line react/prop-types
const NewHeader = ({ children }) => <h1 style={{ color: 'red' }}>TEST {children}</h1>;
...

addParameters({
  docs: {
    components: { h1: NewHeader },
  },
});
```

**Comment by Elanhant** [comment]
I've been experiencing the same issue, and it seems like it's happening in [Documentation-only MDX](https://github.com/storybookjs/storybook/blob/next/addons/docs/docs/mdx.md#documentation-only-mdx), MDX with stories seems to be working fine. 

Perhaps a related problem — docs theming is not applied to Documentation-only MDX, still works for other docs though.

**Comment by shilman** [comment]
@Elanhant thanks for the clue. Can you try upgrading to 6.0-alpha? Both theming & component overrides seem to be working in `next` for me. Have not tried in `master`.

**Comment by Elanhant** [comment]
> @Elanhant thanks for the clue--very interesting... Can you try upgrading to 6.0-alpha? Both theming & component overrides seem to be working in `next` for me. Have not tried in `master`.

Sorry, I should've mention that I am already on alpha 🙂 I've just tried upgrading to `6.0.0-alpha.28`, still no luck. 

I also remember that "page" parameter wasn't working for me either. It's as if doc-only MDX just skips all the theming and parameters, and goes straight to displaying plain MDX.

**Comment by shilman** [comment]
@Elanhant what are the filenames of your docs-only MDX files?

**Comment by Elanhant** [comment]
> @Elanhant what are the filenames of your docs-only MDX files?

`Intro.stories.mdx` and `About.stories.mdx`. [You can check them out here](https://github.com/gatsby-inc/gatsby-interface/tree/master/__stories__/about).

**Comment by shilman** [comment]
Thanks @Elanhant -- am able to repro. It's getting late here, but I'll dig in tomorrow and see if I can figure out what's going on. That's NOT expected behavior.

**Comment by shilman** [comment]
@Elanhant It's something wrong with your repo.

```
yarn why react
=> Found "react@16.13.1"
yarn why @emotion/core
=> Found "@emotion/core@10.0.28"
```

When I started, I found multiple versions of these libraries and aggressively removed stuff, esp. old version of `@storybook/addon-info` and `react-storybook-addon-chapters`, **removed the lockfile**, reinstalled, and generally stripped things down (`FileUpload` stopped building at some point, so I stripped it out). I also removed your workaround of `<MdxDocsTheme />` to test.

In the end this is what I got working in `preview.js`:

```js
const NewHeader = ({ children }) => (
  <h1 style={{ color: "red" }}>TEST {children}</h1>
)

export const parameters = {
  viewport: { viewports },
  docs: {
    components: { h1: NewHeader },
    theme: themes.dark,
  },
}
```

I didn't stop to figure out exactly which change fixed things, but if I had to guess, it would be that there were multiple copies of React and that a value was getting injected into one version's Context and then somehow unavailable in the other version.
